x/oracle/client/cli/tx.go

```diff
@@ -208,3 +204,122 @@ where "terra1..." is the address you want to delegate your voting rights to.
+
+// GetCmdAggregateExchangeRatePrevote will create a aggregateExchangeRatePrevote tx and sign it with the given key.
+func GetCmdAggregateExchangeRatePrevote(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "aggregate-prevote [salt] [exchange_rates] [validator]",
+		Args:  cobra.RangeArgs(2, 3),
+		Short: "Submit an oracle aggregate prevote for the exchange rates of Luna",
+		Long: strings.TrimSpace(`
+Submit an oracle aggregate prevote for the exchange rates of Luna denominated in multiple denoms.
+The purpose of aggregate prevote is to hide aggregate exchange rate vote with hash which is formatted 
+as hex string in SHA256("{salt}:{exchange_rate}{denom},...,{exchange_rate}{denom}:{voter}")
+
+# Aggregate Prevote
+$ terracli tx oracle aggregate-prevote 1234 8888.0ukrw,1.243uusd,0.99usdr 
+
+where "ukrw,uusd,usdr" is the denominating currencies, and "8888.0,1.243,0.99" is the exchange rates of micro Luna in micro denoms from the voter's point of view.
+
+If voting from a voting delegate, set "validator" to the address of the validator to vote on behalf of:
+$ terracli tx oracle prevote 1234 8888.0ukrw,1.243uusd,0.99usdr terravaloper1...
+`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			salt := args[0]
+			exchangeRatesStr := args[1]
+			_, err := types.ParseExchangeRateTuples(exchangeRatesStr)
+			if err != nil {
+				return fmt.Errorf("given exchange_rates {%s} is not a valid format; exchange_rate should be formatted as DecCoins", exchangeRatesStr)
+			}
+
+			// Get from address
+			voter := cliCtx.GetFromAddress()
+
+			// By default the voter is voting on behalf of itself
+			validator := sdk.ValAddress(voter)
+
+			// Override validator if validator is given
+			if len(args) == 3 {
+				parsedVal, err := sdk.ValAddressFromBech32(args[2])
+				if err != nil {
+					return errors.Wrap(err, "validator address is invalid")
+				}
+				validator = parsedVal
+			}
+
+			hash := types.GetAggregateVoteHash(salt, exchangeRatesStr, validator)
+
+			msg := types.NewMsgAggregateExchangeRatePrevote(hash, voter, validator)
+			err = msg.ValidateBasic()
+			if err != nil {
+				return err
+			}
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+
+	return cmd
+}
+
+// GetCmdAggregateExchangeRateVote will create a aggregateExchangeRateVote tx and sign it with the given key.
+func GetCmdAggregateExchangeRateVote(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "aggregate-vote [salt] [exchange_rates] [validator]",
+		Args:  cobra.RangeArgs(2, 3),
+		Short: "Submit an oracle vote for the exchange_rates of Luna",
+		Long: strings.TrimSpace(`
+Submit a vote for the exchange_rates of Luna w.r.t the input denom. Companion to a prevote submitted in the previous vote period. 
+
+$ terracli tx oracle vote 1234 8888.0ukrw,1.243uusd,0.99usdr 
```

Comment:
isn't vote supposed to be aggregate-vote here?

Reply:
right

Reply:
Resolved